File: packages/rules/test/fixtures/square/py/missing-signature-verification.negative.py
import os

from flask import Flask, request
from squareup.utilities.webhooks_helper import WebhooksHelper

app = Flask(__name__)


@app.route("/webhooks/square", methods=["POST"])
def square_webhook():
    sig = request.headers.get("X-Square-HmacSha256-Signature", "")
    body = request.get_data().decode("utf-8")
    valid = WebhooksHelper.is_valid_webhook_event_signature(
        body, sig, os.environ["SQUARE_SIGNATURE_KEY"], request.url
    )
    if not valid:
        return "", 403
    return "", 200

File: packages/rules/test/fixtures/square/js/missing-signature-verification.negative.ts
import express from "express";
import { WebhooksHelper } from "square";

const app = express();
app.use(express.raw({ type: "application/json" }));

app.post("/webhooks/square", async (req, res) => {
  const sig = (req.header("X-Square-HmacSha256-Signature") ?? "") as string;
  const url = `https://example.com${req.originalUrl}`;
  const valid = WebhooksHelper.verifySignature(
    (req.body as Buffer).toString("utf8"),
    sig,
    process.env.SQUARE_SIGNATURE_KEY ?? "",
    url,
  );
  if (!valid) return res.status(403).end();
  res.status(200).end();
});

app.listen(3000);
